x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3548,6 +3559,34 @@ export class RulesClient {
         })
       );
     }
+
+    // check for actions using frequency params if the rule has global frequency params defined
+    if (usesGlobalFreqParams) {
+      const actionsWithFrequency = actions.filter((action) => Boolean(action.frequency));
+      if (actionsWithFrequency.length) {
```

Comment:
I am thinking if we are in this case, we should ignore the global frequency params and just use the per-action frequency and delete the global frequency params when we update the saved object.

Reply:
i was going to type something similar :)
Params in an action has the priority, if they exist we should use them, no matter if there is also rule level notify_when or throttle. But since this looks resolved probably there is a conclusion :) Is this approach better or safer?